providers/src/airflow/providers/edge/executors/edge_executor.py

```diff
@@ -86,9 +88,32 @@ def execute_async(
             )
         )
 
-    @provide_session
-    def sync(self, session: Session = NEW_SESSION) -> None:
-        """Sync will get called periodically by the heartbeat method."""
+    def _check_alive_worker(self, session: Session) -> bool:
+        """Reset worker state if heartbeat timed out."""
+        changed = False
+        heartbeat_interval: int = conf.getint("edge", "heartbeat_interval")
+        workers: list[EdgeWorkerModel] = session.query(EdgeWorkerModel).all()
+
+        for worker in workers:
+            connected = 1
+            if not worker.last_update or worker.last_update < (
+                timezone.utcnow() - timedelta(seconds=heartbeat_interval * 5)
+            ):
```

Comment:
Can you please make this selection into the query to push down the filter to the DB level, not calling `all()` on the database?

Reply:
done